### ISSUE TITLE
Rule Proposal: no-useless-rename

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#6058]
From [disallowIdenticalDestructuringNames].

This rule will prevent to write useless renaming.
Original JSCS rule is checking only object patterns, but my proposal is checking object patterns, import declarations, and export declarations.

```js
{
    "no-useless-rename": ["error", {
        "ignoreDestructuring": false,
        "ignoreImport": false,
        "ignoreExport": false
    }]
}
```

**Valid:**

```js
let {foo: bar} = obj;
import * as foo from "foo";
import {foo as bar} from "foo";
export {foo as bar};
export {foo as bar} from "foo";
```

**Invalid:**

```js
let {foo: foo} = obj;
import {foo as foo} from "foo";
export {foo as foo};
export {foo as foo} from "foo";
```

[disallowIdenticalDestructuringNames]: http://jscs.info/rule/disallowIdenticalDestructuringNames

**Comment by doberkofler** [comment]
+1

**Comment by kaicataldo** [comment]
I can champion and would like to write this rule when we get the details ironed out.

**Comment by mysticatea** [comment]
I'm a champion, too.
Personally, I think the code will get more readability if we remove the useless parts `: foo`/`as foo`.

And we had been having (a part of) this warning in `object-shorthand` rule until #5488.

**Comment by mysticatea** [comment]
I added `accepted` label since there are 3 :+1:s in our team.

**Comment by kaicataldo** [comment]
Sounds good - I can start work on this soon.

**Comment by efender** [comment]
> Invalid:
> ```js
> let {foo: foo} = obj;
> import {foo as foo} from "foo";
> export {foo as foo};
> export {foo as foo} from "foo";
> ```

How about re-export default ([lodash issue](https://github.com/lodash/lodash/pull/2400))?
```js
export { default as default } from './lodash.default.js';
```
Is it correct?
